.github/workflows/deploy.yml

```diff
@@ -0,0 +1,69 @@
+name: Deploy
+
+on:
+  workflow_call:
+    inputs:
+      artifact-name:
+        type: string
+        required: true
+        description: "Name of the artifact to deploy"
+      docs-output-prefix:
+        type: string
+        required: true
+        description: "Path prefix where the docs will be generated"
+      dry-run:
+        type: boolean
+        required: true
+        description: "Dry run (doesn't actually publish anything)"
+    secrets:
+      NUGET_API_KEY:
+        required: true
+      BOT_GITHUB_TOKEN:
+        required: true
+
+env:
+  NUGET_SERVER_URL: https://www.nuget.org/
+  NUGET_API_KEY: ${{ secrets.NUGET_API_KEY }}
+
+jobs:
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Download artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ github.event.inputs.artifact-name }}
+          path: /tmp/artifacts
+      - name: Deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          if [[ "${{ github.event.inputs.dry-run || false }}" == "true" ]]; then
```

Comment:
The `|| false` is because Github Actions doesn't really understand booleans. Or required inputs. (cf. my rant in the chat)

Reply:
Good comment. Worth expanding on in the actual file, for future us (not that I think I'll be able to handle updating this in the future)?

Reply:
Yes, that would probably better. I doubt I'll remember this a year from now.

Reply:
Actually, looking at this again, I realize I don't actually need `|| false`. If `dry-run` is undefined, it will just produce `if [[ "" == "true" ]]`, which is fine.